core/src/main/java/org/testcontainers/dockerclient/DockerClientProviderStrategy.java

```diff
@@ -104,6 +121,71 @@ public DockerClient getClient() {
         return dockerClient;
     }
 
+    /**
+     * TODO we should consider moving this to docker-java at some point
+     */
+    @UnstableAPI
+    protected boolean test() {
+        TransportConfig transportConfig = getTransportConfig();
+        URI dockerHost = transportConfig.getDockerHost();
+
+        Callable<Socket> socketProvider;
+        SocketAddress socketAddress;
+        switch (dockerHost.getScheme()) {
+            case "tcp":
+            case "http":
+            case "https":
+                SocketFactory socketFactory = SocketFactory.getDefault();
+                SSLConfig sslConfig = transportConfig.getSslConfig();
+                if (sslConfig != null) {
+                    try {
+                        socketFactory = sslConfig.getSSLContext().getSocketFactory();
+                    } catch (KeyManagementException | UnrecoverableKeyException | NoSuchAlgorithmException | KeyStoreException e) {
+                        log.warn("Exception while creating SSLSocketFactory", e);
+                        return false;
+                    }
+                }
+                socketProvider = socketFactory::createSocket;
+                socketAddress = new InetSocketAddress(dockerHost.getHost(), dockerHost.getPort());
+                break;
+            case "unix":
+            case "npipe":
+                if (!new File(dockerHost.getPath()).exists()) {
+                    log.debug("DOCKER_HOST socket file '{}' does not exist", dockerHost.getPath());
+                    return false;
+                }
+                socketProvider = () -> {
+                    switch (dockerHost.getScheme()) {
+                        case "unix":
+                            return UnixSocket.get(dockerHost.getPath());
+                        case "npipe":
+                            return new NamedPipeSocket(dockerHost.getPath());
+                        default:
+                            throw new IllegalStateException("Unexpected scheme " + dockerHost.getScheme());
+                    }
+                };
+                socketAddress = new InetSocketAddress("localhost", 2375);
+                break;
+            default:
+                log.warn("Unknown DOCKER_HOST scheme {}, skipping the strategy test...", dockerHost.getScheme());
+                return false;
+        }
+
+        try (Socket socket = socketProvider.call()) {
+            Duration timeout = Duration.ofMillis(200);
+            Awaitility.await()
+                .atMost(TestcontainersConfiguration.getInstance().getClientPingTimeout(), TimeUnit.SECONDS)
+                .pollInterval(timeout)
```

Comment:
```suggestion
            Duration interval = Duration.ofMillis(200);
            Awaitility.await()
                .atMost(TestcontainersConfiguration.getInstance().getClientPingTimeout(), TimeUnit.SECONDS)
                .pollInterval(interval)
```

Since it is not the timeout?

Reply:
it is (the purpose is to have the Socket connect timeout and poll at the same interval)